packages/xstate-test/src/TestModel.ts

```diff
@@ -409,6 +410,21 @@ export class TestModel<TState, TEvent extends EventObject> {
   ): TestModelOptions<TState, TEvent> {
     return { ...this.defaultTraversalOptions, ...this.options, ...options };
   }
+
+  public getShortestPathsFrom(
+    paths: Array<TestPath<TState, TEvent>>
+  ): Array<TestPath<TState, TEvent>> {
+    const resultPaths: any[] = [];
+
+    for (const path of paths) {
+      const shortestPaths = this.getShortestPaths({ initialState: path.state });
+      for (const shortestPath of shortestPaths) {
+        resultPaths.push(joinPaths(path, shortestPath));
+      }
```

Comment:
I'm not sure about this but looking at the usage here... it looks like `this.getShortestPaths` won't consider the existing path. This means that the final path might look somewhat different than a path generated from the true initial state to the "destination" state (as the `getShortestPaths` will start with `visited: emptySet`, or similar). Is it a bug or a feature? :p

Reply:
This is a **feature**. The final path should look different from the path generated from the "true initial state" when you specify the "from state" that the algorithm should start at.

In the below example, the shortest path from the initial state to state B is shown in purple, but if we specify an initial state of **Y**, it _should_ take a different shortest path.

![CleanShot 2022-07-21 at 08 40 47](https://user-images.githubusercontent.com/1093738/180215901-4daca892-4210-476c-9f10-05412067ac18.png)

Reply:
Does this make sense?

Reply:
Yes, it makes sense - could we add a test case that would showcase exactly that? It could be associated with a comment above the `expect` call that would clarify this and make it clear that this is a notable difference from the standard usage that is usually not achievable without `joinPaths`